core/cafe_service.go

```diff
@@ -1026,7 +1063,24 @@ func (h *CafeService) handlePubSubContactQuery(pid peer.ID, env *pb.Envelope) (*
 		return nil, nil
 	}
 
-	return h.service.NewEnvelope(pb.Message_CAFE_PUBSUB_CONTACT_QUERY_RES, res, nil, false)
+	renv, err := h.service.NewEnvelope(pb.Message_CAFE_PUBSUB_CONTACT_QUERY_RES, res, nil, false)
+	if err != nil {
+		return nil, err
+	}
+
+	switch query.ResponseType {
```

Comment:
respond here w/ either direct p2p or pubsub, depending on what the caller wants / needs

Reply:
(clients w/o cafes need pubsub because they don't have a public IP address)